book_scraper/book_scraper/spiders/books.py

```diff
@@ -0,0 +1,64 @@
+import scrapy
+from scrapy.http import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    @staticmethod
+    def get_rating(text: str) -> int:
+        number_str = text.split()[-1]
+        number_dict = {"One": 1, "Two": 2, "Three": 3, "Four": 4, "Five": 5}
+        return number_dict.get(number_str, 0)
+
+    def parse(self, response: Response) -> str:
```

Comment:
Fix type annotation all over the code 
https://docs.python.org/3/library/typing.html#typing.Generator